src/App.js

```diff
@@ -0,0 +1,77 @@
+import './App.css';
+import React, { useState, useEffect } from 'react';
+import Sidebar from './components/Sidebar';
+import Editor from './components/Editor';
+import Split from 'react-split';
+import 'react-mde/lib/styles/css/react-mde-all.css';
+import { nanoid } from 'nanoid';
+
+export default function App() {
+  const [notes, setNotes] = useState(() => JSON.parse(localStorage.getItem('notes')) || []);
+  const [currentNoteId, setCurrentNoteId] = useState((notes[0] && notes[0].id) || '');
+
+  useEffect(() => {
+    localStorage.setItem('notes', JSON.stringify(notes));
+  }, [notes]);
+
+  function createNewNote() {
+    const newNote = {
+      id: nanoid(),
+      body: '# This Is a Title',
+    };
+    setNotes(prevNotes => [newNote, ...prevNotes]);
+    setCurrentNoteId(newNote.id);
+  }
+
+  function updateNote(text) {
+    setNotes(oldNotes => {
+      const newArray = [];
+      for (let i = 0; i < oldNotes.length; i++) {
+        const oldNote = oldNotes[i];
+        if (oldNote.id === currentNoteId) {
+          newArray.unshift({ ...oldNote, body: text });
+        } else {
+          newArray.push(oldNote);
+        }
+      }
+      return newArray;
+    });
+  }
+
```

Comment:
You can make this function shorter and easier to read by mapping, take a look at this suggestion: 
```suggestion
function updateNote(text) {
  setNotes(oldNotes => {
    return oldNotes.map(note => {
      if (note.id === currentNoteId) {
        return { ...note, body: text };
      }
      return note;
    });
  });
}
```